src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -525,20 +530,27 @@ subroutine initialize_ice_shelf_dyn(param_file, Time, ISS, CS, G, US, diag, new_
       enddo ; enddo
       call pass_var(CS%calve_mask,G%domain)
     endif
-
-    call initialize_ice_shelf_boundary_channel(CS%u_face_mask_bdry, CS%v_face_mask_bdry, &
-                CS%u_flux_bdry_val, CS%v_flux_bdry_val, CS%u_bdry_val, CS%v_bdry_val, CS%u_shelf, CS%v_shelf,&
-                CS%h_bdry_val, &
-                CS%thickness_bdry_val, ISS%hmask, ISS%h_shelf, G, &
-                US, param_file )
+   call initialize_ice_shelf_boundary_from_file(CS%u_face_mask_bdry, CS%v_face_mask_bdry, &
+                CS%u_bdry_val, CS%v_bdry_val, CS%umask, CS%vmask, CS%h_bdry_val, &
+                 CS%thickness_bdry_val, ISS%hmask,  ISS%h_shelf, G, US, param_file )
     call pass_var(ISS%hmask, G%domain)
     call pass_var(CS%h_bdry_val, G%domain)
     call pass_var(CS%thickness_bdry_val, G%domain)
     call pass_var(CS%u_bdry_val, G%domain)
     call pass_var(CS%v_bdry_val, G%domain)
     call pass_var(CS%u_face_mask_bdry, G%domain)
     call pass_var(CS%v_face_mask_bdry, G%domain)
+    call pass_var(CS%bed_elev, G%domain)
+    call pass_var(CS%umask, G%domain)
```

Comment:
The pass_var calls for CS%umask and CS%vmask should be combined into a single pass_vector call for both of them, and similarly for CS%u_shelf and CS%v_shelf.

Reply:
With pass_var calls for CS%umask and CS%vmask these fields are correct, 
![image](https://user-images.githubusercontent.com/39838355/116904758-20836880-ac0c-11eb-9506-48e2a0ddb68f.png)
<img width="1252" alt="Screen Shot 2021-05-03 at 12 31 27" src="https://user-images.githubusercontent.com/39838355/116904804-2a0cd080-ac0c-11eb-8622-61f10c4a70c1.png">

but after pass_vector(CS%umask,CS%vmask) call their values are replaced by zeros. Any suggestions how to resolve that are greatly appreciated.